src/core/json/json_object.cc

```diff
@@ -18,8 +18,13 @@ optional<JsonType> JsonFromString(string_view input, PMR_NS::memory_resource* mr
     return false;
   };
 
-  // The maximum allowed JSON nesting depth is 256.
-  const uint32_t json_nesting_depth_limit = 256;
+  // The maximum allowed JSON nesting depth is 64.
+  // The limit was reduced from 256 to 64. This change is reasonable, as most documents contain
+  // no more than 20-30 levels of nesting. In the test case, over 128 levels were used, causing
+  // the parser to enter a long stall due to excessive resource consumption. Even a limit of 128
+  // does not mitigate the issue. A limit of 64 is a sensible compromise.
+  // See https://github.com/dragonflydb/dragonfly/issues/5028
+  const uint32_t json_nesting_depth_limit = 64;
```

Comment:
Default for valkey/redis seems to be 128

I quote 

> A JSON value passed to a command can have a depth of up to 128. If you pass to a command a JSON value that contains an object or an array with a nesting level of more than 128, the command returns an error.

I don't have a strong opinion on this. Just leaving this comment here for completeness.

Reply:
yeah, they use a differrent implementation for json. 64 is fine for all sane usecases